lib/ain-evm/src/txqueue.rs

```diff
@@ -263,9 +274,20 @@ impl TransactionQueue {
     }
 
     pub fn remove_txs_by_sender(&self, sender: H160) {
+        let tx_queue = self.get_cloned_vec();
         let mut data = self.data.lock().unwrap();
-        data.transactions.retain(|(tx, _)| {
-            let tx_sender = match tx {
+        for item in tx_queue {
+            let tx_sender = match &item.queue_tx {
+                QueueTx::SignedTx(tx) => tx.sender,
+                QueueTx::BridgeTx(tx) => tx.sender(),
+            };
+            if tx_sender == sender {
+                data.total_fees -= item.tx_fee;
+                data.total_gas_used -= item.gas_used;
+            }
+        }
+        data.transactions.retain(|item| {
+            let tx_sender = match &item.queue_tx {
                 QueueTx::SignedTx(tx) => tx.sender,
                 QueueTx::BridgeTx(tx) => tx.sender(),
```

Comment:
We could do the fee/gas reduction in the retain loop. Shouldn't be required to clone the vec and iterate twice

Reply:
Resolved in 0d3f42934